src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -104,6 +95,47 @@ namespace cryptonote
 #endif
     block_reward += fee;
 
+    const bool do_carrot = hard_fork_version >= HF_VERSION_CARROT;
+    if (do_carrot)
+    {
+      try
+      {
+        carrot::CarrotDestinationV1 miner_destination;
```

Comment:
Nit: I'd probably put all this in a helper function like `make_single_enote_carrot_coinbase_transaction_v1` or something similar to shrink the size of this function. I think it's nice that each of these individual functions exist too to make it easier on p2pool or whomever